src/BaseCollection.php

```diff
@@ -128,7 +128,7 @@ public function chunk(int $length): array
 		}
 
 		return array_map(function ($array) {
-			return new static($array);
+			return new self($array);
```

Comment:
This is wrong, as it will return a `BaseCollection` instance when an inherited class instance is expected instead. If the return type hint allows for `static`, then it should be used instead.

Reply:
fixed